src/multivariate/mvnormalcanon.jl

```diff
@@ -100,28 +104,29 @@ function MvNormalCanon(J::P) where P<:AbstractPDMat
     MvNormalCanon{eltype(J),P,ZeroVector{eltype(J)}}(z, z, J)
 end
 
-function MvNormalCanon(h::Vector{T}, J::P) where {T<:Real, P<:AbstractPDMat}
+function MvNormalCanon(h::AbstractVector{T}, J::P) where {T<:Real, P<:AbstractPDMat}
     length(h) == dim(J) || throw(DimensionMismatch("Inconsistent argument dimensions"))
     R = Base.promote_eltype(h, J)
-    hh, JJ = convert(AbstractArray{R}, h), convert(AbstractArray{R}, J)
+    hh, JJ = collect(convert(AbstractArray{R}, h)), convert(AbstractArray{R}, J)
```

Comment:
Was this collect needed somewhere?

Reply:
In case `h` is `I` or `ZeroVector`?

Reply:
yes, some linear algebra operations are not defined on `AbstractArrays` I think